library/entropy.c

```diff
@@ -271,7 +271,13 @@ static int entropy_gather_internal( mbedtls_entropy_context *ctx )
      */
     for( i = 0; i < ctx->source_count; i++ )
     {
-        if( ctx->source[i].strong == MBEDTLS_ENTROPY_SOURCE_STRONG )
+        volatile int strong_fi = ctx->source[i].strong;
+        if( strong_fi == MBEDTLS_ENTROPY_SOURCE_STRONG )
```

Comment:
I think the double-check here is the wrong way round: if the attacker can glitch one of the two checks, they've succeeded. I think here we want nested `if`s, so that the attacker can only succeed by glitching two checks in a row (which is very unlikely with the random delay in-between).